dlmalloc_nonreuse.c

```diff
@@ -3842,7 +3855,7 @@ static mchunkptr try_realloc_chunk(mstate m, mchunkptr p, size_t nb) {
         newp = p;
       }
     }
-    else if(!cinuse(next)) {
+    else if(!cinuse(next) && !cdirty(next)) {
```

Comment:
Upon reading the code further myself, this should instead be an assertion. When quarantining, freed memory first goes into freebin until a revocation sweep is done, at which point the real free implementation, ie `dlfree_internal`, is called, which is the point where we mess with the inuse flags.

I have verified that `free` followed by `realloc` does not reuse memory, but inserting a `malloc_revoke` in between does allow the memory to be reused. But I have to say the current flags implementation is a mess and I don't quite understand how it all fits together.